esmvaltool/cmorizers/data/formatters/datasets/esacci_aerosol.py

```diff
@@ -0,0 +1,282 @@
+"""ESMValTool CMORizer for ESACCI-AEROSOL data.
+
+Tier
+   Tier 2: other freely-available dataset.
+
+Source
+   CCI CEDA ftp: ftp://anon-ftp.ceda.ac.uk/neodc/esacci/aerosol/data/
+       ATSR2_SU/L3/v4.3/MONTHLY/ (1997-2002)
+       AATSR_SU/L3/v4.3/MONTHLY/ (2003-2011)
+       ATSR2_SU/L3/v4.3/DAILY/ (1997-2002)
+       AATSR_SU/L3/v4.3/DAILY/ (2003-2011)
+   Other years are not considered since they are not complete.
+
+
+Last access
+   20240522
+
+Download and processing instructions
+   Download the following files:
+       ftp: ftp://anon-ftp.ceda.ac.uk/neodc/esacci/aerosol/data/
+           ATSR2_SU/L3/v4.3/MONTHLY/YYYY/*.nc
+           AATSR_SU/L3/v4.3/MONTHLY/YYYY/*.nc
+           ATSR2_SU/L3/v4.3/DAILY/YYYY/MM/*.nc
+           AATSR_SU/L3/v4.3/DAILY/YYYY/MM/*.nc
+   and put all monthly files into one directory named {version}-monthly
+   all daily files into one directory named {version}-daily.
+   {version} is defined in cmorizers/data/cmor_config/ESACCI-AEROSOL.yml
+   (e.g. version: 'SU-v4.3')
+
+Modification history
+   20240522-lauer_axel: written.
+"""
+
+import glob
+import logging
+import os
+from copy import deepcopy
+from datetime import datetime
+from dateutil import relativedelta
+
+import cf_units
+import iris
+import numpy as np
+from dask import array as da
+from esmvalcore.cmor.table import CMOR_TABLES
+
+from ...utilities import save_variable
+
+logger = logging.getLogger(__name__)
+
+
+def _create_nan_cube(cube, year, month, day):
+    """Create cube containing only nan from existing cube."""
+    nan_cube = cube.copy()
+    nan_cube.data = da.ma.masked_greater(cube.core_data(), -1e20)
+
+    # Read dataset time unit and calendar from file
+    dataset_time_unit = str(nan_cube.coord('time').units)
+    dataset_time_calender = nan_cube.coord('time').units.calendar
+    # Convert datetime
+    newtime = datetime(year=year, month=month, day=day)
+    newtime = cf_units.date2num(newtime, dataset_time_unit,
+                                dataset_time_calender)
+    nan_cube.coord('time').points = float(newtime)
+
+    return nan_cube
```

Comment:
note however that those values are not NaNs in the strict data type sense, they are `numpy.Masked` objects (that look like `--` and have actual values of `fill_value` which I think it's 1.e+20 here)